packages/connect-core/src/protocol-connect/headers.spec.ts

```diff
@@ -0,0 +1,177 @@
+// Copyright 2021-2023 Buf Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Utility function to compare two header objects
+export function compare(h1: Headers, h2: Headers): boolean {
```

Comment:
I'm not convinced this is a good idea. It makes the tests difficult to read, because you have to reason about this function first.

What do you think about a helper like this (we're already this it elsewhere)?

```ts
function listHeaderKeys(header: Headers): string[] {
  const keys: string[] = [];
  header.forEach((_, key) => keys.push(key));
  return keys;
}
```

Reply:
I like it.  Done.